src/main/java/io/fabric8/maven/docker/StartMojo.java

```diff
@@ -483,6 +488,45 @@ private String getTcpHost(WaitConfiguration.TcpConfiguration tcpConfig, Properti
         return host;
     }
 
+    private WaitUtil.WaitChecker getHealthyWaitChecker(final String imageConfigDesc, final DockerAccess docker, final String containerId, final List<String> logOut) {
```

Comment:
More a note to myself: We should extract functionality out of `StartMojo` as it already grows quite a bit. The wait stuff could go into an extra package (without normal instead of inner classes).

Also easier to test then.